### PR TITLE
CLN: type annotations

### PR DESCRIPTION
**Pull request by jbrockmendel**
Separating these out from the non-CLN work going on in/around these functions.

### CODE REVIEW

File: pandas/core/groupby/ops.py [hunk 1]
```diff
@@ -790,7 +790,7 @@ def _get_axes(group):
         return group.axes
 
 
-def _is_indexed_like(obj, axes):
+def _is_indexed_like(obj, axes) -> bool:
```
Comment (WillAyd) [inline pandas/core/groupby/ops.py]: I think reasonable to annotate `obj` as FrameOrSeries here, unless the TypeVar doesn't work in which case a Union

**Review by WillAyd — CHANGES_REQUESTED** [review]
Just as a general comment I think it would be better to fully annotate some methods rather than partially annotating a lot of methods. The real value in doing this is getting the hard ones reviewed; can always have newer contributors sprint on easier annotations

**Comment by jbrockmendel** [comment]
> Just as a general comment I think it would be better to fully annotate some methods rather than partially annotating a lot of methods.

That's reasonable.  How strong is this preference?  i.e. when weighed against "its nice to do these separately from the other PR that makes non-typing changes in the same functions to have a simpler diff"

**Comment by jreback** [comment]
> > Just as a general comment I think it would be better to fully annotate some methods rather than partially annotating a lot of methods.
> 
> That's reasonable. How strong is this preference? i.e. when weighed against "its nice to do these separately from the other PR that makes non-typing changes in the same functions to have a simpler diff"

I would rather merge quickly on correct changes that are non-controversial, rather block things. so actually happy to type the 'easy' things (of course nice for newcomers to do this too).

**Comment by WillAyd** [comment]
FWIW I’m not blocking on this just have reviewed thoroughly. Feel free to merge if happy with it

File: pandas/core/reshape/tile.py [hunk 2]
```diff
@@ -498,13 +503,15 @@ def _convert_bin_to_datelike_type(bins, dtype):
...
+def _format_labels(
+    bins, precision, right: bool = True, include_lowest: bool = False, dtype=None
+):
     """ based on the dtype, return our labels """
 
     closed = "right" if right else "left"
 
     if is_datetime64tz_dtype(dtype):
-        formatter = partial(Timestamp, tz=dtype.tz)
+        formatter = lambda x: Timestamp(x, tz=dtype.tz)
```
Comment (WillAyd) [inline pandas/core/reshape/tile.py]: What error was this throwing? I would think a partial to be preferable to a lambda

Comment (jbrockmendel) [reply pandas/core/reshape/tile.py]: `error: Incompatible types in assignment (expression has type "Callable[[Any], Any]", variable has type "partial[Any]")`

Comment (WillAyd) [reply pandas/core/reshape/tile.py]: Hmm that's unfortunate. I suppose goes back to https://github.com/python/mypy/issues/1484

**Review by WillAyd — APPROVED** [review]
lgtm @jreback or @simonjayhawkins

**Comment by jbrockmendel** [comment]
green

**Comment by jbrockmendel** [comment]
gentle ping.  a couple of upcoming branches are going to have merge conflicts with this otherwise

**Comment by jreback** [comment]
thanks

**Pull request merged**